### DISCUSSION TITLE
GNU screen Not Compiling On Linux

### DISCUSSION THREAD

**Discussion by tundratim** [Homebrew/discussions#1389] [Everyday usage]
I am attempting to build GNU screen from source for installation in a custom location on a Linux machine and am seeing the error below.  Guidance appreciated:

```
utoheader: WARNING: Using the third argument of 'AC_DEFINE_UNQUOTED' and
autoheader: WARNING: 'AC_DEFINE' allows one to define a template without
autoheader: WARNING: 'acconfig.h':
autoheader: 
autoheader: WARNING:   AC_DEFINE([NEED_FUNC_MAIN], 1,
autoheader: 		[Define if a function 'main' is needed.])
autoheader: 
autoheader: WARNING: More sophisticated templates can also be produced, see the
autoheader: WARNING: documentation.
configure.ac: error: required file 'install-sh' not found
configure.ac:   try running autoreconf --install

==> Formula
Path: /opt/TundraWare/tools/Library/Taps/homebrew/homebrew-core/Formula/screen.rb
```

**Comment by gromgit** [comment]
What's the output of `brew config` and `brew doctor`?

**Comment by carlocab** [comment]
Try making this change to the formula:
```diff
diff --git a/Formula/screen.rb b/Formula/screen.rb
index 2484e8d32af..4892c8a25c5 100644
--- a/Formula/screen.rb
+++ b/Formula/screen.rb
@@ -45,6 +45,7 @@ class Screen < Formula
     # https://savannah.gnu.org/bugs/index.php?59465
     ENV.append "CFLAGS", "-Wno-implicit-function-declaration"
 
+    system "autoreconf", "-fiv"
     system "./autogen.sh"
     system "./configure", "--prefix=#{prefix}",
                           "--mandir=#{man}",
```

**Comment by tundratim** [comment]
On 4/30/21 6:59 PM, Carlo Cabrera wrote:
> Try making this change to the formula:
> 
> diff --git a/Formula/screen.rb b/Formula/screen.rb
> index 2484e8d32af..4892c8a25c5 100644
> --- a/Formula/screen.rb
> +++ b/Formula/screen.rb
> @@ -45,6 +45,7 @@ class Screen < Formula
>      # https://savannah.gnu.org/bugs/index.php?59465
>      ENV.append "CFLAGS", "-Wno-implicit-function-declaration"
>  
> + system "autoreconf", "-fiv"
>      system "./autogen.sh"
>      system "./configure", "--prefix=#{prefix}",
>                            "--mandir=#{man}",


Yes, this seems to have fixed it.  How do we get this change into
master?

**Reply by carlocab** [reply]
Open a pull request (or two). You'll probably need to start with a PR at https://github.com/Homebrew/linuxbrew-core so that the change can be tested on Linux CI, and then another PR at https://github.com/Homebrew/homebrew-core so that it can be tested with macOS CI and merged if everything works fine.

You can open both PRs simultaneously. (It's a little complicated now because Linuxbrew is--slowly--being merged into homebrew/core. It'll be a lot simpler when that's finished.)

Please make sure to read the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md), in particular the parts about commit style in Homebrew.
